job-webhook/job_scraper.py

```diff
@@ -4,69 +4,99 @@
 Scrapes job postings from GitHub repository and sends updates via Discord webhook
 """
 
+import dataclasses
 import hashlib
 import json
 import logging
 import os
 import time
-from dataclasses import dataclass
 from datetime import datetime
 from typing import Optional
 
-import dataclasses
 import requests
 from bs4 import BeautifulSoup
 from dotenv import load_dotenv
 
 load_dotenv()
-MAGIC_LITERAL = "\u21b3"  # Magic literal for company name continuation
+COMPANY_NAME_CONTINUATION = "\u21b3"
 
 # Configure logging
 logging.basicConfig(
     level=logging.INFO,
-    format='%(asctime)s - %(levelname)s - %(message)s',
-    handlers=[
-        logging.FileHandler('job_scraper.log'),
-        logging.StreamHandler()
-    ]
+    format="%(asctime)s - %(levelname)s - %(message)s",
+    handlers=[logging.FileHandler("job_scraper.log"), logging.StreamHandler()],
 )
 logger = logging.getLogger(__name__)
 
-@dataclass
+
+@dataclasses.dataclass
 class JobPosting:
     """Represents a job posting with all relevant information."""
-    
+
     company: str
     role: str
     location: str
     application_link: str
     date_posted: str
-    
-    def to_dict(self) -> dict:
-        """Convert job posting to dictionary format."""
-        return {
-            'company': self.company,
-            'role': self.role,
-            'location': self.location,
-            'application_link': self.application_link,
-            'date_posted': self.date_posted
-        }
+
+
+@dataclasses.dataclass
+class JobCache:
+    """Represents cached job postings and content hash."""
+
+    content_hash: str
+    jobs: list[JobPosting]
+    last_updated: str
+
+
+@dataclasses.dataclass
+class EmbedField:
+    """Represents a field in a Discord embed."""
+
+    name: str
+    value: str
+    inline: bool = False
+
+
+@dataclasses.dataclass
+class EmbedFooter:
+    """Represents the footer of a Discord embed."""
+
+    text: str
+    icon_url: Optional[str] = None
+
+
+@dataclasses.dataclass
+class DiscordEmbed:
+    """Represents a Discord embed structure."""
+
+    title: str
+    description: str
+    color: int
+    fields: list[EmbedField]
+    footer: EmbedFooter
+    timestamp: str
+
 
 class JobScraper:
     """Main class for scraping job postings and sending Discord notifications."""
-    
-    def __init__(self, webhook_url: str, github_url: str, 
-                 cache_file: str = "job_cache.json"):
+
+    def __init__(
+        self, webhook_url: str, github_url: str, cache_file: str = "job_cache.json"
```

Comment:
Prefer using [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html) instead of raw `str`s when dealing with file paths. Same rationale as the "stringly-typed antipattern" discussion above. There's also a lot neater to work with in general.

```suggestion
        self, webhook_url: str, github_url: str, cache_file: Path = Path("job_cache.json")
```

Once you do that, you should also change all call sites of `open(string_path)` to its more idiomatic pattern:

```diff
-with open(string_path, encoding="utf-8") as f:
+with path_obj.open(encoding="utf-8") as f:
```